chrome/content/zotero/xpcom/style.js

```diff
@@ -238,24 +238,36 @@ Zotero.Styles = new function() {
 	/**
 	 * Installs a style file, getting the contents of an nsIFile and showing appropriate
 	 * error messages
-	 * @param {String|nsIFile} style An nsIFile representing a style on disk, or a string
-	 *     containing the style data
+	 * @param {Object} style An object with one of the following properties
+	 * 		- file: An nsIFile representing a style on disk
+	 * 		- url: A url of the location of the style (local or remote)
+	 * 		- string: A string containing the style data
 	 * @param {String} origin The origin of the style, either a filename or URL, to be
 	 *     displayed in dialogs referencing the style
 	 * @param {Boolean} [silent=false] Skip prompts
 	 */
 	this.install = Zotero.Promise.coroutine(function* (style, origin, silent=false) {
 		var styleTitle;
+		var warnDeprecated;
+		if (style instanceof Components.interfaces.nsIFile) {
+			warnDeprecated = true;
+			style = {file: style};
+		} else if (Object.prototype.toString.call(style) === "[object String]") {
+			warnDeprecated = true;
+			style = {string: style};
+		}
+		if (warnDeprecated) {
+			Zotero.debug("Zotero.Styles.install() now takes a style object as first argument -- update your code", 2);
+		}
 		
 		try {
-			if (style instanceof Components.interfaces.nsIFile) {
-				// handle nsIFiles
-				var url = Services.io.newFileURI(style);
-				var xmlhttp = yield Zotero.HTTP.request("GET", url.spec);
-				styleTitle = yield _install(xmlhttp.responseText, style.leafName, false, silent);
-			} else {
-				styleTitle = yield _install(style, origin, false, silent);
+			if (style.file) {
+				style.url = Services.io.newFileURI(style.file).spec;
```

Comment:
Should add a `style = Zotero.File.pathToFile(style);` above this line so that `file` can be a string path, which would be the preferred way to pass a file going forward. (And should update the function documentation above.)